### ISSUE TITLE
Bug: use-isnan does not detect globalThis.NaN

### ISSUE THREAD

**Issue by Kuldeep2822k** [eslint/eslint#20854]
### Environment

ESLint Playground

### What parser are you using?

Default (Espree)

### Playground Code
```js
if (foo === globalThis.NaN) {
    console.log("This will never run");
}

let index = [1, 2, 3].indexOf(globalThis.NaN);
```

### ESLint Configuration 
<details> 
 <summary> Tap to see full config </summary> 

```js
import { defineConfig } from "eslint/config";

export default defineConfig([
    {
        "rules": {
            "use-isnan": [
                "error",
                {
                    "enforceForIndexOf": true
                }
            ]
        }
    }
]);
```
 </details>

### What did you expect to happen?

An error should be reported for both cases, warning that comparisons to `NaN` are always false, and recommending the use of `Number.isNaN()`.

ESLint currently reports errors correctly for the exact equivalent expressions:
* `foo === NaN`
* `foo === Number.NaN` 

`globalThis.NaN` is the exact same built-in global `NaN` value, accessed via the standardized ES2020 static reference. Because globalThis.NaN === foo suffers from the exact same language pitfall (NaN === NaN will always evaluate to false), it should be tracked symmetrically to Number.NaN.

### What actually happened?

No errors are reported. `globalThis.NaN` is completely ignored by the rule, resulting in a false negative for users who explicitly prefix their global references.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiaWYgKGZvbyA9PT0gZ2xvYmFsVGhpcy5OYU4pIHtcbiAgICBjb25zb2xlLmxvZyhcIlRoaXMgd2lsbCBuZXZlciBydW5cIik7XG59XG5cbmxldCBpbmRleCA9IFsxLCAyLCAzXS5pbmRleE9mKGdsb2JhbFRoaXMuTmFOKTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7InVzZS1pc25hbiI6WyJlcnJvciIseyJlbmZvcmNlRm9ySW5kZXhPZiI6dHJ1ZX1dfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
This behavior is not specific to `globalThis` as the rule only checks for `NaN` and `Number.NaN` (and not accesses via other global objects such as `global.NaN`, `window.NaN`, ...).
I agree that this is a false negative and fixing this can be seen as a bug rather than a feature (which would be breaking) but I'd like additional input from the team.
I could not find related discussion to this.

**Comment by Tanujkanti4441** [comment]
I agree that `globalThis.NaN` should be reported in these cases.

>this is a false negative and fixing this can be seen as a bug rather than a feature (which would be breaking)

In that case, it think for now, we can add an option to enable the reporting of `globalThis.NaN` or we can just update the docs to ask users to use `no-restricted-globals` rule to disallow `globalThis.NaN` ([playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcmVzdHJpY3RlZC1nbG9iYWxzOiBbXCJlcnJvclwiLCB7IGdsb2JhbHM6IFtcIk5hTlwiXSwgY2hlY2tHbG9iYWxPYmplY3Q6IHRydWUgfV0qL1xuXG5pZiAoZm9vID09PSBnbG9iYWxUaGlzLk5hTikge1xuICAgIGNvbnNvbGUubG9nKFwiVGhpcyB3aWxsIG5ldmVyIHJ1blwiKTtcbn1cblxubGV0IGluZGV4ID0gWzEsIDIsIDNdLmluZGV4T2YoZ2xvYmFsVGhpcy5OYU4pOyIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==))?

@eslint/eslint-team thoughts?

**Comment by Kuldeep2822k** [comment]
I’m checking in on this as it’s been a couple of weeks since the last discussion. Has the team reached a consensus on whether this should be treated as a bug fix or a new feature

**Comment by kecrily** [comment]
I think it would be better to add it as an option, and perhaps make it the default behavior in the next major release.

**Comment by idkned** [comment]
hi is this issue open to work on

**Comment by DMartens** [comment]
No as the team has not yet reached a decision and as such is not marked as accepted.
You can read more about the process [here](https://eslint.org/docs/latest/contribute/work-on-issue).
